pkg/reconciler/reconciler.go

```diff
@@ -525,7 +529,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		return ctrl.Result{}, err
 	}
 
-	rel, state, err := r.getReleaseState(actionClient, obj, vals.AsMap())
+	rel, state, releaseStateErr := r.getReleaseState(actionClient, obj, vals.AsMap())
+	release := release.Release{}
+	if rel != nil {
+		release = *rel
+	}
+
+	if obj.GetDeletionTimestamp() != nil {
```

Comment:
This can't be moved here. If you are in a state where `r.getValues` returns an error (and this might happen simply from a misconfiguration at the CR level, such as referencing a secret that doesn't exist, resulting in us not being able to build the values due to missing secret content), you will find the controller to be unable to perform a deletion. Deletions must strictly be handled _before_ any non-internal error (such as not being able to get the action client) might terminate the control flow.

Reply:
Get it. I guess this would then imply that the PreDeletion extension should neither receive `vals`, nor `release`, right?

[On the other hand, one could also introduce a fallback mechanism where `vals` is simply passed as empty to the `PreDelete` extension in case an error occurs while assembling it -- similar to the situation with `release`].